### PR TITLE
DEPR: `null_color` since `color` used everywhere else

### PR DESCRIPTION
**Pull request by attack68**
in all the builtin highlighters `color` is used. makes this more consistent.

### CODE REVIEW

**Comment by jreback** [comment]
no objections, prob this is showing a warning in another test

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -3133,8 +3144,12 @@ def highlight_null(
         def f(data: DataFrame, props: str) -> np.ndarray:
             return np.where(pd.isna(data).to_numpy(), props, "")
 
+        if null_color != "red":
+            raise FutureWarning("`null_color` is deprecated: use `color` instead")
```
Comment (mroeschke) [inline pandas/io/formats/style.py:3148]: Shouldn't this use `warning.warn(...)`?

Also needs a whatsnew note for 1.5.0

Comment (attack68) [reply pandas/io/formats/style.py]: agreed chged

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -3133,8 +3144,16 @@ def highlight_null(
         def f(data: DataFrame, props: str) -> np.ndarray:
             return np.where(pd.isna(data).to_numpy(), props, "")
 
+        if null_color != "red":
+            warnings.warn(
+                "`null_color` is deprecated: use `color` instead",
```
Comment (mroeschke) [inline pandas/io/formats/style.py:3149]: Could you add a test for this warning?

Comment (attack68) [reply pandas/io/formats/style.py:3149]: done and green

**Review by mroeschke — APPROVED** [review]

File: pandas/io/formats/style.py [hunk 3]
```diff
@@ -3133,8 +3144,16 @@ def highlight_null(
         def f(data: DataFrame, props: str) -> np.ndarray:
             return np.where(pd.isna(data).to_numpy(), props, "")
 
+        if null_color != "red":
```
Comment (jreback) [inline pandas/io/formats/style.py:3147]: you need to make `null_color` == `lib.no_default` here (and then if its not that you can warn)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 3 above)
Comment (attack68) [reply pandas/io/formats/style.py]: done

**Review by jreback — APPROVED** [review]

**Pull request merged**